packages/synapse-react-client/src/components/CreateTableViewWizard/CreateTableViewWizard.integration.test.tsx

```diff
@@ -0,0 +1,793 @@
+import React from 'react'
+import { act, render, screen, waitFor, within } from '@testing-library/react'
+import { server } from '../../mocks/msw/server'
+import CreateTableViewWizard, {
+  CreateTableViewWizardProps,
+} from './CreateTableViewWizard'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import mockProjectEntityData from '../../mocks/entity/mockProject'
+import userEvent from '@testing-library/user-event'
+import {
+  ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+  ENTITY_VIEW_TYPE_MASK_FILE,
+  ENTITY_VIEW_TYPE_MASK_FOLDER,
+  ENTITY_VIEW_TYPE_MASK_PROJECT,
+  EntityType,
+  MATERIALIZED_VIEW_CONCRETE_TYPE_VALUE,
+  SUBMISSION_VIEW_CONCRETE_TYPE_VALUE,
+  TABLE_ENTITY_CONCRETE_TYPE_VALUE,
+  VIRTUAL_TABLE_CONCRETE_TYPE_VALUE,
+} from '@sage-bionetworks/synapse-types'
+import { addColumnModelToForm } from '../TableColumnSchemaEditor/TableColumnSchemaEditorTestUtils'
+import SynapseClient from '../../synapse-client'
+import {
+  MOCK_ACCESS_TOKEN,
+  MOCK_CONTEXT_VALUE,
+} from '../../mocks/MockSynapseContext'
+import { EntityFinderModal } from '../EntityFinder/EntityFinderModal'
+import defaultFileViewColumnModels from '../../mocks/query/defaultFileViewColumnModels'
+import { rest } from 'msw'
+import { BackendDestinationEnum } from '../../utils/functions'
+import { getEndpoint } from '../../utils/functions/getEndpoint'
+import { MOCK_ANNOTATION_COLUMNS } from '../../mocks/mockAnnotationColumns'
+import { mockEvaluationQueue } from '../../mocks/entity/mockEvaluationQueue'
+
+jest.mock('../EntityFinder/EntityFinderModal', () => ({
+  EntityFinderModal: jest.fn(() => (
+    <div data-testid="EntityFinderModalMocked" />
+  )),
+}))
+
+const mockEntityFinderModal = jest.mocked(EntityFinderModal)
+
+jest.spyOn(SynapseClient, 'createEntity')
+jest.spyOn(SynapseClient, 'createColumnModels')
+jest.spyOn(SynapseClient, 'getAnnotationColumnModels')
+
+const createColumnModelsSpy = jest.mocked(SynapseClient.createColumnModels)
+const createEntitySpy = jest.mocked(SynapseClient.createEntity)
+const getAnnotationColumnsSpy = jest.mocked(
+  SynapseClient.getAnnotationColumnModels,
+)
+describe('CreateTableWizard integration tests', () => {
+  function renderComponent(props: CreateTableViewWizardProps) {
+    return render(<CreateTableViewWizard {...props} />, {
+      wrapper: createWrapper({
+        ...MOCK_CONTEXT_VALUE,
+        isInExperimentalMode: true,
+      }),
+    })
+  }
+
+  beforeAll(() => {
+    server.listen()
+  })
+  afterEach(() => {
+    server.resetHandlers()
+    jest.clearAllMocks()
+  })
+  afterAll(() => server.close())
+
+  test('Create a table', async () => {
+    const onComplete = jest.fn()
+    const onCancel = jest.fn()
+    renderComponent({
+      open: true,
+      parentId: mockProjectEntityData.id,
+      onComplete,
+      onCancel,
+    })
+
+    // Select "Table"
+    const tableButton = await screen.findByRole('menuitem', {
+      name: 'Table',
+    })
+    await userEvent.click(tableButton)
```

Comment:
Based on this recommendation in the [docs](https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent), could consider setting up the user via `const user = userEvent.setup()` before the component is rendered, rather than relying on `userEvent.click` to call `setup` internally